models/articles.model.js

```diff
@@ -25,31 +25,44 @@ exports.selectArticles = () => {
     });
 };
 
-const checkArticleIdExists = (article_id) => {
+exports.selectCommentsByArticleId = (article_id) => {
   return db
     .query("SELECT * FROM articles WHERE article_id = $1;", [article_id])
     .then((data) => {
-      return data.rows.length !== 0;
+      if (data.rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          message: "No such article_id yet",
+        });
+      }
+      return db
+        .query(
+          "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;",
+          [article_id]
+        )
+        .then((data) => {
+          return data.rows;
+        });
     });
 };
-exports.selectCommentsByArticleId = (article_id) => {
+
+exports.insertCommentById = (article_id, username, body) => {
   return db
-    .query(
-      "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC;",
-      [article_id]
-    )
+    .query("SELECT * FROM articles WHERE article_id = $1;", [article_id])
```

Comment:
You shouldn't need this query, if you try to insert an article where the id doesn't exist then you should get a psql error which will be handled by your psql error handler